### ISSUE TITLE
Реализовать или хотя бы вшить авто-подбор параметров

### ISSUE THREAD

**Issue by Gerbariy1** [ValdikSS/GoodbyeDPI#709]
### I've made sure there's no existing feature request / Я убедился, что такой функциональности еще никто не предлагал

- [X] I've made sure there's no existing feature request / Я убедился, что такой функциональности еще никто не предлагал

### Describe your feature / Опишите ваше предложение

Увидел решение: https://app-time.ru/post/instrukciya-po-avtomaticheskomu-podboru-nastroek-dlya-goodbyedpi, очень понравилось. Хотел бы видеть в этом репозитории

**Comment by shikulja** [comment]
GBDPI занимается фильтрацией трафика. Автоподбор касается доменом и хост листов в первую очередь, и там свои танцы с бубном. Тем более что проектов по автоподборке уже 4.
Тот что вы написали уже устарел.
Примерный список: blockcheck(zapret), GoodCheck by Ori, goodcheckgo by Ori, DPI_Blockcheck

**Comment by Gerbariy1** [comment]
Конечно, гораздо лучше ждать фикс на форумах по типу хабра и прочих, чем включить разок программку и подобрать **оптимальнейшие** для твоего региона аргументы. Ну а раз их уже целых 4 и они постоянно устаревают, то тем более стоит сделать свой, официальный, а самое главное рабочий автоподбор

**Comment by MayanKoyote** [comment]
Зачем-то лучше свои мозги в голове иметь чем фикс на форумах на типа хабра и прочих ждать.

**Comment by Anizam** [comment]
> Тем более что проектов по автоподборке уже 4.

Можно было бы и назвать парочку, чтобы порекламировать хорошие проекты

**Comment by Capewearer** [comment]
> Зачем-то лучше свои мозги в голове иметь чем фикс на форумах типа хабра и прочих ждать.

Наш узколобый товарищ, далеко не все имеют непосредственное отношение к IT сфере, особенно к её административной части. И далеко не все имеют волю, компетенцию и желание ставить фиксы из десятка источников, совершая монотонный цикл установки, проверки и удаления нерабочего фикса.

**Comment by Amegatron** [comment]
> И далеко не все имеют волю, компетенцию и желание ставить фиксы из десятка источников

Давайте, главное, не забывать:
1) Первопричину того, почему мы здесь. Это наиглавнейшее. Не автор данного проекта придумал блокировать все на свете и мешать людям жить.
2) Данный проект никому ничем не обязан. Это лишь инструмент для совершения вполне определенных действий. Кто хочет/умеет пользоваться - тот пользуется.
3) Один проект не может (и не должен) включать в себя сразу все, что приходит в голову пользователям. Тем более за бесплатно.
4) Решение данной проблемы (автоподбор параметров) может быть осуществлено сторонними средствами поверх GBDPI. Если они нерабочие или неактуальные, можете попробовать (нет) предъявить претензии их авторам.

**Comment by Nok1a** [comment]
С помощью питона можно руинить  гудбай, бровзер и перебирать параметры. Единственное что приходится ручками отмечать проигрывание видео или не проигрывание Гудбай запускается с помощью батника, просто пишите скрипт с перебором параметров , им изменяете батник, вызываете его, вызываете видос любой, смотрите как работают или не работают, если нет , то тыкаете нет , всё руинится и запускается заного и так пока не оживёт, ожило жмёте ДА параметры записываются и последующие запуски будут с ними. 
Я пока такое решение нашёл. 

А вообще попробуйте это чем изобретать.

"@ECHO OFF
PUSHD "%~dp0"
set _arch=x86
IF "%PROCESSOR_ARCHITECTURE%"=="AMD64" (set _arch=x86_64)
IF DEFINED PROCESSOR_ARCHITEW6432 (set _arch=x86_64)
PUSHD "%_arch%"

start "" goodbyedpi.exe -e 2 —fake-from-hex 1603030135010001310303424143facf5c983ac8ff20b819cfd634cbf5143c0005b2b8b142a6cd335012c220008969b6b387683dedb4114d466ca90be3212b2bde0c4f56261a9801 -q —native-frag —set-ttl 4 —fake-gen 5

POPD
POPD"

**Comment by prblm** [comment]
> С помощью питона можно руинить гудбай, бровзер и перебирать параметры. Единственное что приходится ручками отмечать проигрывание видео или не проигрывание Гудбай запускается с помощью батника, просто пишите скрипт с перебором параметров , им изменяете батник, вызываете его, вызываете видос любой, смотрите как работают или не работают, если нет , то тыкаете нет , всё руинится и запускается заного и так пока не оживёт, ожило жмёте ДА параметры записываются и последующие запуски будут с ними. Я пока такое решение нашёл.
> 
> А вообще попробуйте это чем изобретать.
> 
> "@echo OFF PUSHD "%~dp0" set _arch=x86 IF "%PROCESSOR_ARCHITECTURE%"=="AMD64" (set _arch=x86_64) IF DEFINED PROCESSOR_ARCHITEW6432 (set _arch=x86_64) PUSHD "%_arch%"
> 
> start "" goodbyedpi.exe -e 2 —fake-from-hex 1603030135010001310303424143facf5c983ac8ff20b819cfd634cbf5143c0005b2b8b142a6cd335012c220008969b6b387683dedb4114d466ca90be3212b2bde0c4f56261a9801 -q —native-frag —set-ttl 4 —fake-gen 5
> 
> POPD POPD"
> 
> Если есть умные челики, которые знают как проверять завелись видосы или нет, то скрипт перебора и генерации HEX можно сделать абсолютно автоматическим. имея скачаный гудбай и питон. Мне просто впадлу искать пакеты которые текут при загрузке и т.д, если есть люди которые уже понимают о чем я спрашиваю...напишите)))) Забацаю перебор)

для проверки , что сработали параметры можно тестировать скорость отдачи ютуба командой:

`curl --connect-to ::speedtest.selectel.ru -o/dev/null -k https://manifest.googlevideo.com/100MB`
